electrumsv/nodeapi.py

```diff
@@ -892,6 +902,42 @@ async def jsonrpc_getnewaddress_async(request: web.Request, request_id: RequestI
         output_script)
     return script_template_to_string(script_template)
 
+
+async def jsonrpc_getrawchangeaddress_async(request: web.Request, request_id: RequestIdType,
+        parameters: RequestParametersType) -> Any:
+    """
+    Reserve the next unused change address (otherwise known as external key) and return it as
+    a P2PKH address. This differs from the getnewaddress endpoint in that there is no blockchain
+    monitoring for this address.
+
+    Raises `HTTPInternalServerError` for related errors to return to the API using application.
+    """
+    # Ensure the user is accessing either an explicit or implicit wallet.
+    wallet = get_wallet_from_request(request, request_id)
+    assert wallet is not None
+
+    # Similarly the user must only have one account (and we will ignore any
+    # automatically created petty cash accounts which we do not use yet).
+    accounts = wallet.get_visible_accounts()
+    if len(accounts) != 1:
+        raise web.HTTPInternalServerError(headers={ "Content-Type": "application/json" },
+            text=json.dumps(ResponseDict(id=request_id, result=None,
+                error=ErrorDict(code=RPCError.WALLET_ERROR,
+                    message=f"Ambiguous account (found {len(accounts)}, expected 1)"))))
+    account = accounts[0]
+    key_data = account.reserve_unassigned_key(CHANGE_SUBPATH, KeyInstanceFlag.IS_RAW_CHANGE_ADDRESS)
```

Comment:
One of the reasons I have started putting "Raises `SomeError` ..." in method documentation is that otherwise when you come back to the method, you have to trace all the way down the execution in all the functions they call, and work it out again everytime you visit the function and need to know what can get raised out of it and what you should handle.

This can raise a databaseupdate error out uncaught.